plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -47,6 +56,43 @@ def __init__(
             **kwargs,
         )
 
+    def _connect_to_duckdb(self, provider: DuckDBProvider, hosted_secret: Optional[Secret]):
+        """
+        Handles the connection to DuckDB based on the provider.
+
+        Args:
+            provider: DuckDB provider
+            hosted_secret: Secret containing authentication token for the hosted provider
+
+        Returns:
+            A DuckDB connection object.
+        """
+        connection_map = {
```

Comment:
instead of creating a map, you could use `functools.partial`: https://www.linkedin.com/pulse/how-enumerate-functions-python-fan-wu/